treetime/distribution.py

```diff
@@ -224,14 +225,12 @@ def xmax(self):
 
 
     def __call__(self, x):
-
         if isinstance(x, Iterable):
             valid_idxs = (x > self._xmin-TINY_NUMBER) & (x < self._xmax+TINY_NUMBER)
-            res = np.ones_like (x, dtype=float) * (BIG_NUMBER+self.peak_val)
-            tmp_x = np.copy(x[valid_idxs])
-            tmp_x[tmp_x<self._xmin+TINY_NUMBER] = self._xmin+TINY_NUMBER
-            tmp_x[tmp_x>self._xmax-TINY_NUMBER] = self._xmax-TINY_NUMBER
-            res[valid_idxs] = self._peak_val + self._func(tmp_x)
+            res = np.full(np.shape(x), BIG_NUMBER+self.peak_val, dtype=float)
+            tmp_x = x[valid_idxs]
+            np.clip(tmp_x, self._xmin+TINY_NUMBER, self._xmax-TINY_NUMBER)
+            res[valid_idxs] = self._peak_val + np.interp(tmp_x, self.xvals, self.yvals)
```

Comment:
what is the advantage of reinstantiating `interp` over the initial instance in line 153 above? it this is faster, one could get rid of the above, right? (while dropping support for anything but linear interpolation)